sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+import reactor.core.publisher.Mono;
+import com.azure.core.http.HttpHeader;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionAsyncClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionAsyncClientBuilder
+ */
+public class HttpUrlConnectionAsyncClient implements HttpClient {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpUrlConnectionAsyncClient.class);
+    private final Duration connectionTimeout;
+    private final ProxyOptions proxyOptions;
+    private final Configuration configuration;
+
+    HttpUrlConnectionAsyncClient(Duration connectionTimeout, ProxyOptions proxyOptions, Configuration configuration) {
+        this.connectionTimeout = connectionTimeout;
+        this.proxyOptions = proxyOptions;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
+    }
+
+    /**
+     * Asynchronously send the HttpRequest
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param context The context of the request, for any additional changes
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendAsync(HttpRequest httpRequest, Context context) {
+        ProgressReporter progressReporter = Contexts.with(context).getHttpRequestProgressReporter();
+        HttpMethod httpMethod = httpRequest.getHttpMethod();
+        if (httpMethod == HttpMethod.PATCH) {
+            return sendPatchViaSocket(httpRequest);
+        }
+        return handleConnection(httpRequest, progressReporter);
+    }
+
+    /**
+     * Send a PATCH request via a SocketClient
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @return A Mono containing a HttpResponse object
+     */
+    private Mono<HttpResponse> sendPatchViaSocket(HttpRequest httpRequest) {
+        return Mono.fromCallable(() -> SocketClient.sendPatchRequest(httpRequest));
+    }
+
+    /**
+     * Open a connection based on the HttpRequest URL
+     *
+     * @param httpRequest The HTTP Request being sent
+     * @param progressReporter (Optional) for reporting progress while writing the request body
+     * @return A Mono containing a HttpUrlConnectionResponse object
+     */
+    private Mono<HttpResponse> handleConnection(HttpRequest httpRequest, ProgressReporter progressReporter) {
```

Comment:
Let's break this method down into smaller chunks of logic, my initial thought on the split is

- connect
- sendRequest
- receiveResponse